### DISCUSSION TITLE
[@types/node] Several missing members from `crypto` module and `WebCrypto` object

### DISCUSSION THREAD

**Discussion by jhmaster2000** [DefinitelyTyped/DefinitelyTyped#60511] [Issues with a @types package] [answered]
**Note:** I am now using the list below to track my progress in implementing those for a PR.
The following members are missing from the `node:crypto` module: (Parens are the typings' major node versions each type belongs to)
- [x] [`crypto.DiffieHellmanGroup`](https://nodejs.org/docs/latest/api/crypto.html#class-diffiehellmangroup) (**all**)
- [x] [`crypto.createDiffieHellmanGroup`](https://nodejs.org/docs/latest/api/crypto.html#cryptocreatediffiehellmangroupname) (**all**)
- [x] [`crypto.setEngine`](https://nodejs.org/docs/latest/api/crypto.html#cryptosetengineengine-flags) (**all**)
- [x] [`crypto.setFips`](https://nodejs.org/docs/latest/api/crypto.html#cryptosetfipsbool) (**all**)
- [x] [`crypto.getRandomValues`](https://nodejs.org/docs/latest/api/crypto.html#cryptogetrandomvaluestypedarray) (**latest**)
- [x] [`crypto.subtle`](https://nodejs.org/docs/latest/api/crypto.html#cryptosubtle) (**latest**)
- [ ] ~~`crypto.Cipheriv` and `crypto.Cipheriv` - No docs found, but seem to be just aliases to `crypto.Cipher` and `crypto.Decipher`? (**?**)~~ (I will not be doing these as I don't feel comfortable doing them without any docs)

While [`crypto.webcrypto`](https://nodejs.org/docs/latest/api/crypto.html#cryptowebcrypto) is present, virtually all of it's members are missing as follows: (Since `crypto.webcrypto` was added on v15, everything on it is **v16**+**latest**, there are no exceptions currently)
- [x] [`crypto.webcrypto.subtle`](https://nodejs.org/docs/latest/api/webcrypto.html#cryptosubtle)
- [x] [`crypto.webcrypto.getRandomValues`](https://nodejs.org/docs/latest/api/webcrypto.html#cryptogetrandomvaluestypedarray)
- [x] [`crypto.webcrypto.randomUUID`](https://nodejs.org/docs/latest/api/webcrypto.html#cryptorandomuuid)
- [x] [`crypto.webcrypto.CryptoKey`](https://nodejs.org/docs/latest/api/webcrypto.html#class-cryptokey)
- [x] [`crypto.webcrypto.CryptoKeyPair`](https://nodejs.org/docs/latest/api/webcrypto.html#class-cryptokeypair)
- [x] More not listed here have been done, they are listed on the linked PR now.

A lot more **WebCrypto** classes exist which are not exposed to `crypto.webcrypto` but should still be documented at ambient level for use on parameters and return types of other functions that make use of them, such as [`CryptoKeyPair`](https://nodejs.org/docs/latest/api/webcrypto.html#class-cryptokeypair).

Is there any particular reason for all of these missing things? Most of these are extremely old too, dating back to Node v0.x.x, with the most recent ones being Node v17.4.x. Am I missing some hidden reason for all of the above not being present on `@types/node`?

### Update:
I have now made a PR for adding these here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60778
Progress updates will now take place in that PR.

[a comment by an automated account is not shown]

**Comment by mcollina** [comment, marked as answer]
There are no reason for those to not be present minus maintainer time. Would you like to send a PR?

**Reply by jhmaster2000** [reply]
I'd happy to try and make a PR for these, are there any special guidelines for `@types/node` I should follow or be aware of? Or would I be fine just following the DT guidelines for all packages?

**Reply by peterblazejewicz** [reply]
I'd say: README is fine, start small (pick one that you're accustomed to). There is a lot of versioning in @types/node. But there is no hurry, your contribution is welcome!
